File: backend/Models/Product.cs
using System.ComponentModel.DataAnnotations;

namespace backend.Models
{
    public class Product
    {
        public int Id { get; set; }
        
        [Required]
        [MaxLength(200)]
        public string Name { get; set; } = string.Empty;
        
        [Required]
        [MaxLength(50)]
        public string Sku { get; set; } = string.Empty;
        
        // Backward-compat string category label for display/filtering
        [Required]
        [MaxLength(100)]
        public string Category { get; set; } = string.Empty;

        // Strong FK to Category for relational integrity (optional to preserve legacy data)
        public int? CategoryId { get; set; }
        public Category? CategoryRef { get; set; }
        
        [MaxLength(1000)]
        public string? Description { get; set; }
        
        [Required]
        [Range(0, double.MaxValue)]
        public decimal Price { get; set; }
        
        [Range(0, double.MaxValue)]
        public decimal? OriginalPrice { get; set; }
        
        [Required]
        [Range(0, int.MaxValue)]
        public int Stock { get; set; }
        
        [Required]
        [Range(0, double.MaxValue, ErrorMessage = "Khối lượng phải lớn hơn hoặc bằng 0 gram")]
        public decimal Weight { get; set; } // Khối lượng sản phẩm (gram)
        
        [Required]
        [MaxLength(20)]
        public string Status { get; set; } = "active"; // active, inactive
        
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
        
        // Navigation properties
        public ICollection<ProductImage> Images { get; set; } = new List<ProductImage>();
        public ICollection<ProductColor> Colors { get; set; } = new List<ProductColor>();
        public ICollection<ProductSticker> Stickers { get; set; } = new List<ProductSticker>();
    }
}

File: frontend/src/types/product.ts
export interface ProductImage {
  id: number
  imageUrl: string
  sortOrder: number
  isPrimary: boolean
  colorCode?: string
}

export interface ProductColor {
  id: number
  colorCode: string
  colorName?: string
  sortOrder: number
}

export interface ProductSticker {
  id: number
  imageUrl: string
  sortOrder: number
}

export interface Product {
  id: number
  name: string
  sku: string
  category: string
  description?: string
  // Regular product fields (nullable for custom products)
  price?: number
  originalPrice?: number
  stock?: number
  weight?: number
  primaryWarehouseId?: string
  primaryWarehouseName?: string
  // Custom product fields (nullable for regular products)
  consultationNote?: string
  // Common fields
  status: 'active' | 'inactive'
  createdAt: string
  updatedAt: string
  images: ProductImage[]
  colors: ProductColor[]
  stickers?: ProductSticker[]
}

export interface CreateProductRequest {
  name: string
  sku: string
  category: string
  description?: string
  // Regular product fields (optional for custom products)
  price?: number
  originalPrice?: number
  stock?: number
  weight?: number
  primaryWarehouseId?: string
  // Custom product fields (optional for regular products)
  consultationNote?: string
  // Common fields
  status: 'active' | 'inactive'
  colors?: string[]
  imageUrls?: string[]
  imageFiles?: File[]
  // Note: Backend tự động map ảnh với màu theo thứ tự upload
}
